Adding CONTRIBUTING.md

Djelibeybi opened the pull request:
Provide documented contribution rules and guidelines to assist new users on how best to structure their Dockerfile or updates for the quickest review/merge process.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

gvenzl commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,72 @@
+1. Fork this repository
+2. Create a branch in your fork to implement the changes. We recommend using
+the issue number as part of your branch name, e.g. `1234-fixes`
+3. Ensure that any documentation is updated with the changes that are required
```
There should be another point saying something like "Ensure that samples are updated if the PR causes the behavior of the sample to change"

gvenzl commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,72 @@
+
+1. Do not run `ssh` inside a container.
+1. Do not use host networking mode (`--net=host`) for a container.
+1. Do not hard-code any passwords.
```
This should probably elaborate on that it's best practice to auto generate a password by default via openssl rand and accept a password as argument.

gvenzl commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,72 @@
+- Don't install any interactive/user tools, e.g. things like `vim`, `less` or
+`man`. Debugging should be done prior to the image submission.
+- Don't install `wget` as the base images already include `curl`.
+- Always remember to run `yum clean all` in the same `RUN` directive as a
```
Include that the image should in general be kept as small as possible and unnecessary packages/parts removed.

Djelibeybi commented:
I've pushed some changes. Great feedback, thanks @gvenzl. Can you review again, please?

brunoborges commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,78 @@
+# Contributing
+Oracle welcomes contributions to this repository from anyone.
+
+If you want to submit a pull request to fix an bug or enhance an existing
```
"fix a bug".

brunoborges commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,78 @@
+
+We have some golden rules that we require all submitted `Dockerfiles` to abide
+by. These rules are provided by Oracle Global Product Security and may change
+at any time.
```
These rules are of course applied to all Oracle employees submitting contributions, but they will be _checked_ by us when contributions are coming from 3rd-parties. Maybe the wording here should be more directed to employees?

Djelibeybi replied:
No, these rules apply to both employees and non-employees alike. We will not accept any PRs that don't follow these rules.

nirmay commented:
looks good to me. Thanks for adding the rules.

Djelibeybi commented:
Excellent feedback, @mriccell.  I've added that note and added some text around documenting any defaults that are used in the case that custom values are not provided.

Djelibeybi commented:
Thanks team. Merged. If you have any other guidelines/rules, feel free to open a PR to include those.
